lib/puppet/provider/cert_file/posix.rb

```diff
@@ -0,0 +1,63 @@
+require 'pp'
+
+Puppet::Type.type(:cert_file).provide :posix do
+  confine feature: :posix
+
+  def exists?
+    return false unless Pathname.new(resource[:path]).exist?
```

Comment:
What happens if the path exists and is a directory?

Reply:
In that case it will fail later and raise a proper Puppet error.